packages/lib-classifier/src/store/FeedbackStore.spec.js

```diff
@@ -1,150 +1,196 @@
-        T1: [{
-          id: 'testRule2-1',
-          hideSubjectViewer: true,
-          answer: '0',
-          strategy: 'testStrategy',
-          success: false,
-          successEnabled: true,
-          successMessage: 'Yippee!',
-          failureEnabled: true,
-          failureMessage: 'Nope!'
-        }]
-      },
-      showModal: false
-    }
-    feedback = FeedbackStore.create(feedbackStub)
-  })
-
-  it('should exist', function () {
-    expect(FeedbackStore).to.be.ok()
-    expect(FeedbackStore).to.be.an('object')
+  
+  describe('existance', function () {
+    it('should exist', function () {
+      expect(FeedbackStore).to.be.ok()
+      expect(FeedbackStore).to.be.an('object')
+    })
   })
 
-  describe('createRules', function () {
-    const project = {
-      id: '1'
-    }
-    const workflow = {
-      id: '2'
-    }
-    const subject = {
-      id: '3'
-    }
+  describe('Actions > createRules', function () {
+    let workflow, project, subject, feedback, feedbackStub
+    before(function () {
+      workflow = WorkflowFactory.build()
+      project = ProjectFactory.build({}, { activeWorkflowId: workflow.id })
+      subject = Factory.build('subject')
+      feedbackStub = FeedbackFactory.build()
+      feedback = FeedbackStore.create(feedbackStub)
+      feedback.projects = ProjectStore.create()
+      feedback.workflows = WorkflowStore.create()
+      feedback.subjects = SubjectStore.create()
+    })
 
     beforeEach(function () {
-      feedback.projects = {
-        active: project
-      }
-      feedback.workflows = {
-        active: workflow
-      }
-      feedback.createRules(subject)
+      feedback.projects.setResource(project)
+      feedback.projects.setActive(project.id)
+      feedback.workflows.setResource(workflow)
+      feedback.workflows.setActive(workflow.id)
     })
 
     afterEach(function () {
+      feedback.reset()
       helpers.isFeedbackActive.resetHistory()
       helpers.generateRules.resetHistory()
     })
 
     it('should set active state', function () {
-      expect(helpers.isFeedbackActive).to.have.been.calledOnceWith(project, subject, workflow)
+      expect(feedback.isActive).to.be.false()
+      feedback.createRules(subject)
+      expect(helpers.isFeedbackActive).to.have.been.calledOnce()
+      // Normally we would use sinon's withArgs call to test the spy for specific arguments
+      // But this is complicated with the fact that the arguments actually used in the store are MST references
+      // so they will not equal the stub resource object
+      // We convert the arguments to standard json for the expectations below
+      // Expect for subject
+      expect(helpers.isFeedbackActive.args[0][0].toJSON()).deep.equal(project)
+      expect(helpers.isFeedbackActive.args[0][1]).deep.equal(subject)
+      expect(helpers.isFeedbackActive.args[0][2].toJSON()).deep.equal(workflow)
+      expect(feedback.isActive).to.equal(helpers.isFeedbackActive.returnValues[0])
```

Comment:
Maybe code this to `true` or `false`, depending on what we expect from the mock. I'm assuming it's supposed to set `isActive` to true.

If `isActive` is false here, that could be because it was false at the beginning of the test.

Reply:
Before this test didn't actually test that `isFeedbackActive` was setting `self.isActive` to its return value. The store was instantiated with `isActive` as true, the stubbed function was just returning `feedback.isActive`, so it didn't make sense based on the `it` statement since the store and stubbed function was always true and referring to itself with no explicit and testable change. 

I've updated this so that the store is instantiated with `isActive` as false and tests for it, the stubbed function specifically returns `true`, and test that the function's return value is now set to `isActive`. Now this is testing what the `it` statement says it is testing. 

I can add another expectation that `isActive` is true, but I'll leave this one as is because we want to be sure that `isActive` is set to the return value since that's what the code is doing.

Reply:
OK, so if I follow correctly, then if the code is broken so that `isFeedbackActive()` returns false for these rules, this test will pass even though the store is broken?

Reply:
I would say that in that scenario that it is `isFeedbackActive` is what is broken and its own tests would ideally fail, not that the store is broken. The code sets `isActive` to whatever the return value of `isFeedbackActive`, which could be the same as what `isActive` is already set to. It doesn't check prior to setting it that it is not equal to the current setting. Maybe that should change in a future update and then this test would cover that too?

For the purposes of the test though, I think adding another expectation here that `isActive` is true is a good idea to make it explicit that the store is being updated with the return value of `isFeedbackActive` that we expect, but I don't think it's a good idea to replace the return value expectation.